docs/DeveloperGuide.md

```diff
@@ -6,8 +6,77 @@
 
 ## Design & implementation
 
+# High Level Functionalities
+![HighLevel.png](HighLevel.png)
```

Comment:
Are the arrows standard?
![image](https://github.com/user-attachments/assets/8fc1e66f-5634-4f89-815a-b9257ac5626f)